File: 0045-jump-game-ii/0045-jump-game-ii.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var jump = function(nums) {
    let result = 0;
        
        for (let i = 0, target = 0, maxDistance = nums[0]; i <= target; ++i) {
            if (target >= nums.length - 1) break;
            maxDistance = Math.max(maxDistance, i + nums[i]);
            
            if (i == target) {
                ++result;
                target = maxDistance;
            }
        }
        return result;
};

File: 0045-jump-game-ii/0045-jump-game-ii.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        int result = 0;

        for (int i = 0, target = 0, maxDistance = nums[0]; i <= target; ++i) {
            if (target >= nums.size() - 1) break;
            maxDistance = max(maxDistance, i + nums[i]);

            if (i == target) {
                ++result;
                target = maxDistance;
            }
        }
        return result;
    }
};